### ISSUE TITLE
What do the 3 and 2 (args[1] and args[2]) in the circle mean?

### ISSUE THREAD

**Issue by jiyuwangbupt** [ultralytics/yolov5#1729]
## ❔Question
![image](https://user-images.githubusercontent.com/37127150/102599914-1e3ad900-4159-11eb-8f3b-b0800bb526ef.png)
What do the 3 and 2 (args[1] and args[2]) in the circle mean?

**Comment by glenn-jocher** [comment]
@jiyuwangbupt see module for module arguments.

**Comment by jiyuwangbupt** [comment]
> @jiyuwangbupt see module for module arguments.

![image](https://user-images.githubusercontent.com/37127150/102737116-82d87c80-4381-11eb-8354-b544c0470ae1.png)
I did not see the specific description of the module, just gave an args array. I understand that the first parameter of args is the number of convolution kernels, the second is the size of the convolution kernel, and the third is the multiple of the size reduction after the feature map passes. Is this the case?

**Comment by glenn-jocher** [comment]
@jiyuwangbupt Conv() module is defined in models/common.py.
https://github.com/ultralytics/yolov5/blob/394d1c89f33f29b2039c54f4a81e561a0b140b3a/models/common.py#L26-L39

**Comment by jiyuwangbupt** [comment]
![image](https://user-images.githubusercontent.com/37127150/103088682-82144480-4626-11eb-900b-941ddd3b3b51.png)
![image](https://user-images.githubusercontent.com/37127150/103088766-d3bccf00-4626-11eb-8b4d-26bceb1945d7.png)
Hello, there are 6 parameters in the conv module you gave, but there are only 3 args in yaml, how does this correspond? The first one I understand should be the number of convolution kernels, the second is the kernel, and the second is the stride?

**Comment by glenn-jocher** [comment]
@jiyuwangbupt input channel count c1 is automatically tracked, so the 3 arguments are argument indices 1-3, or c2, k, s.

**Comment by jiyuwangbupt** [comment]
I modified YOLOv5 head in yolov5m.yaml and reported an error halfway through the training.










------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ultralytics/yolov5"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年1月30日(星期六) 上午8:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ultralytics/yolov5] What do the 3 and 2 (args[1] and args[2]) in the circle mean? (#1729)





 
Closed #1729.

**Comment by glenn-jocher** [comment]
@jiyuwangbupt 👋 Hello, thank you for your interest in our work! This issue seems to lack the minimum requirements for a proper response, or is insufficiently detailed for us to help you. Please note that most technical problems are due to:

- **Your modified or out-of-date code**. If your issue is not reproducible in a new `git clone` version of this repo we can not debug it. Before going further run this code and verify your issue persists:
```bash
$ git clone https://github.com/ultralytics/yolov5 yolov5_new  # clone latest
$ cd yolov5_new
$ python detect.py  # verify detection

# CODE TO REPRODUCE YOUR ISSUE HERE
```

- **Your custom data.** If your issue is not reproducible in one of our 3 common datasets ([COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [COCO128](https://github.com/ultralytics/yolov5/blob/master/data/coco128.yaml), or [VOC](https://github.com/ultralytics/yolov5/blob/master/data/voc.yaml)) we can not debug it. Visit our [Custom Training Tutorial](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data) for guidelines on training your custom data. Examine `train_batch0.jpg` and `test_batch0.jpg` for a sanity check of your labels and images.

- **Your environment.** If your issue is not reproducible in one of the verified environments below we can not debug it. If you are running YOLOv5 locally, verify your environment meets all of the [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies specified below. If in doubt, download Python 3.8.0 from https://www.python.org/, create a new [venv](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/), and install requirements.

If none of these apply to you, we suggest you close this issue and raise a new one using the 🐛 **Bug Report template**, providing screenshots and a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) of your issue. Thank you!


## Requirements

Python 3.8 or later with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed, including `torch>=1.7`. To install run:
```bash
$ pip install -r requirements.txt
```


## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>


## Status

![CI CPU testing](https://github.com/ultralytics/yolov5/workflows/CI%20CPU%20testing/badge.svg)

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training ([train.py](https://github.com/ultralytics/yolov5/blob/master/train.py)), testing ([test.py](https://github.com/ultralytics/yolov5/blob/master/test.py)), inference ([detect.py](https://github.com/ultralytics/yolov5/blob/master/detect.py)) and export ([export.py](https://github.com/ultralytics/yolov5/blob/master/models/export.py)) on MacOS, Windows, and Ubuntu.

**Comment by jiyuwangbupt** [comment]
I modified the network structure and caused this problem. The original network structure can run without problems.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ultralytics/yolov5"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月15日(星期一) 中午1:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ultralytics/yolov5] What do the 3 and 2 (args[1] and args[2]) in the circle mean? (#1729)





 
@jiyuwangbupt 👋 Hello, thank you for your interest in our work! This issue seems to lack the minimum requirements for a proper response, or is insufficiently detailed for us to help you. Please note that most technical problems are due to:
  
Your modified or out-of-date code. If your issue is not reproducible in a new git clone version of this repo we can not debug it. Before going further run this code and verify your issue persists:
  $ git clone https://github.com/ultralytics/yolov5 yolov5_new  # clone latest $ cd yolov5_new $ python detect.py  # verify detection # CODE TO REPRODUCE YOUR ISSUE HERE
  
 
Your custom data. If your issue is not reproducible in one of our 3 common datasets (COCO, COCO128, or VOC) we can not debug it. Visit our Custom Training Tutorial for guidelines on training your custom data. Examine train_batch0.jpg and test_batch0.jpg for a sanity check of your labels and images.
 
 
 
Your environment. If your issue is not reproducible in one of the verified environments below we can not debug it. If you are running YOLOv5 locally, verify your environment meets all of the requirements.txt dependencies specified below. If in doubt, download Python 3.8.0 from https://www.python.org/, create a new venv, and install requirements.
 
  
If none of these apply to you, we suggest you close this issue and raise a new one using the 🐛 Bug Report template, providing screenshots and a minimum reproducible example of your issue. Thank you!
 
Requirements
 
Python 3.8 or later with all requirements.txt dependencies installed, including torch&gt;=1.7. To install run:
 $ pip install -r requirements.txt
 
Environments
 
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):
  
Google Colab and Kaggle notebooks with free GPU:  
 
Google Cloud Deep Learning VM. See GCP Quickstart Guide
 
Amazon Deep Learning AMI. See AWS Quickstart Guide
 
Docker Image. See Docker Quickstart Guide 
  
Status
 

 
If this badge is green, all YOLOv5 GitHub Actions Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training (train.py), testing (test.py), inference (detect.py) and export (export.py) on MacOS, Windows, and Ubuntu.

**Comment by jiyuwangbupt** [comment]
I modified the network structure and caused this problem. The original network structure can run without problems. This is the modified yolov5x.yaml。When I was running halfway, an error occurred. From the figure, it seems to be the error generated by the cat module during the third GPU training.Could you please help me look at the problem? 





Best!


------------------ 原始邮件 ------------------
发件人:                                                                                                                        "ultralytics/yolov5"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年3月15日(星期一) 中午1:05
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ultralytics/yolov5] What do the 3 and 2 (args[1] and args[2]) in the circle mean? (#1729)





 
@jiyuwangbupt 👋 Hello, thank you for your interest in our work! This issue seems to lack the minimum requirements for a proper response, or is insufficiently detailed for us to help you. Please note that most technical problems are due to:
  
Your modified or out-of-date code. If your issue is not reproducible in a new git clone version of this repo we can not debug it. Before going further run this code and verify your issue persists:
  $ git clone https://github.com/ultralytics/yolov5 yolov5_new  # clone latest $ cd yolov5_new $ python detect.py  # verify detection # CODE TO REPRODUCE YOUR ISSUE HERE
  
 
Your custom data. If your issue is not reproducible in one of our 3 common datasets (COCO, COCO128, or VOC) we can not debug it. Visit our Custom Training Tutorial for guidelines on training your custom data. Examine train_batch0.jpg and test_batch0.jpg for a sanity check of your labels and images.
 
 
 
Your environment. If your issue is not reproducible in one of the verified environments below we can not debug it. If you are running YOLOv5 locally, verify your environment meets all of the requirements.txt dependencies specified below. If in doubt, download Python 3.8.0 from https://www.python.org/, create a new venv, and install requirements.
 
  
If none of these apply to you, we suggest you close this issue and raise a new one using the 🐛 Bug Report template, providing screenshots and a minimum reproducible example of your issue. Thank you!
 
Requirements
 
Python 3.8 or later with all requirements.txt dependencies installed, including torch&gt;=1.7. To install run:
 $ pip install -r requirements.txt
 
Environments
 
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):
  
Google Colab and Kaggle notebooks with free GPU:  
 
Google Cloud Deep Learning VM. See GCP Quickstart Guide
 
Amazon Deep Learning AMI. See AWS Quickstart Guide
 
Docker Image. See Docker Quickstart Guide 
  
Status
 

 
If this badge is green, all YOLOv5 GitHub Actions Continuous Integration (CI) tests are passing. These tests evaluate proper operation of basic YOLOv5 functionality, including training (train.py), testing (test.py), inference (detect.py) and export (export.py) on MacOS, Windows, and Ubuntu.
